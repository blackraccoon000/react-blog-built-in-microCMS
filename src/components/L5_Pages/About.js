import React from 'react';
import styled from 'styled-components';
import cheerio from 'cheerio';
import hljs from 'highlight.js/lib/core';
import javascript from 'highlight.js/lib/languages/javascript';
import xml from 'highlight.js/lib/languages/xml';
import 'highlight.js/styles/atelier-cave-dark.css';
import Loading from '../L1_Atoms/Loading';

hljs.registerLanguage('javascript', javascript);
hljs.registerLanguage('xml', xml);

import ArticleBody from '../L2_Molecules/ArticleBody';
import ArticleImg from '../L2_Molecules/ArticleImg';
import TitleWrapper from '../L3_Organisms/TitleWrapper';

const Wrapper = styled.div`
  background-color: #61b1c8;
`;

const Container = styled.div.attrs((props) => {})`
  align-items: center;
  background-color: #fbfbf9;
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: calc(800rem / 16);
  margin: 0 auto;
`;

const BodyWrapper = styled.div`
  margin: 20px;
`;

const BodyTitle = styled.h1.attrs((props) => {})`
  align-items: center;
  display: flex;
  font-family: 'kan415typos-std';
  font-size: calc(12 / 16) rem;
  text-align: center;
  margin: calc(20rem / 16) 0 calc(5rem / 16);
  padding: calc(8rem / 16);
`;

const TextLine = styled.p.attrs((props) => {})`
  margin-bottom: 10px;
  border-bottom: 1px solid #000;
  padding-bottom: 6px;
`;

const About = (props) => {
  document.title = 'このページについて';
  const width =
    window.parent.screen.width < 800 ? window.parent.screen.width : 800;
  return (
    <Wrapper>
      <Container>
        <ArticleImg
          flag={false}
          src={`https://images.microcms-assets.io/assets/577bcb7965bf4dbd951686143646657b/c24d720c28a140eb93ff0bbb2c763569/mokuro.png?w=${width}`}
          imageMargin="0 auto"
          width="100%"
        />
        <BodyTitle>このページについて</BodyTitle>
        <BodyWrapper>
          <TextLine>
            このページはReactの練習用ページとして公開しています。
          </TextLine>
          <TextLine>React &amp; Reduxをベースに、</TextLine>
          <TextLine>
            CSSはCSS in JSのstyled-componentsを活用しています。
          </TextLine>
          <TextLine>
            デプロイ先はNetlifyを活用し、コンテンツ管理はMicroCMSをベースにしています。
          </TextLine>
          <TextLine>
            本来、作成したページの詳細は公開しないことが常識ですが、
          </TextLine>
          <TextLine>勉強用ということで全部公開しようと考えています。</TextLine>
          <TextLine>詳しくはまた追記します。</TextLine>
        </BodyWrapper>
      </Container>
    </Wrapper>
  );
};

export default About;
