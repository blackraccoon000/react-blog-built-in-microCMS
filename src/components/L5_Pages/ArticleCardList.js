import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import pages from '../../tests/fixtures/pages';

import ArticleCardTurquoiseBlue from '../L3_Organisms/ArticleCardTurquoiseBlue';

const Wrapper = styled.div.attrs((props) => {
  return {
    style: {
      backgroundColor: `${props.bGColor}`,
    },
  };
})``;

const ArticleCardWrapper = styled.div.attrs((props) => {})`
  display: grid;
  grid-template-rows: auto auto;
  grid-template-columns: 1fr;

  @media screen and (min-width: 400px) and (max-width: 766px) {
    grid-template-columns: 1fr 1fr;
  }

  @media screen and (min-width: 767px) and (max-width: 1024px) {
    grid-template-columns: 1fr 1fr 1fr;
  }

  @media screen and (min-width: 1025px) {
    grid-template-columns: 1fr 1fr 1fr 1fr;
  }
`;

const ArticleCardList = (props) => {
  document.title = 'PlayWell | よく遊びよく学べ';
  return (
    <Wrapper bGColor={props.bGColor}>
      <ArticleCardWrapper>
        {props.views.viewArticles.map((article) => {
          return (
            <ArticleCardTurquoiseBlue
              key={article.id}
              src={article.thumbnail !== undefined ? article.thumbnail.url : ''}
              title={article.title}
              id={article.id}
              createdAt={article.createdAt}
              updatedAt={article.updatedAt}
              keywords={
                article.keyword !== undefined
                  ? article.keyword.split(',')
                  : ['']
              }
            />
          );
        })}
      </ArticleCardWrapper>
    </Wrapper>
  );
};

ArticleCardList.propTypes = {
  bGColor: PropTypes.string,
  articles: PropTypes.array,
  views: PropTypes.object,
  viewArticles: PropTypes.array,
};

ArticleCardList.defaultProps = {
  bGColor: '#61b1c8',
  articles: pages,
  views: {
    availableViews: 4,
    acquisition: 11,
    viewsSortByIds: [
      'uhrjbmf8xqz3',
      'kf9j-jzze',
      'q0qtwov5f4q6',
      'ww2vpab1815b',
    ],
    viewArticles: pages,
  },
};

const mapStateToProps = (state) => {
  const { articles, views, status } = state;
  return {
    articles,
    views,
    status,
  };
};

// const mapDispatchToProps = (dispatch, ownProps) => {
//   return {
//     acquiredUpdater: (props) => acquiredUpdater(dispatch, props),
//   };
// };

export default connect(mapStateToProps)(ArticleCardList);
export { ArticleCardList };
