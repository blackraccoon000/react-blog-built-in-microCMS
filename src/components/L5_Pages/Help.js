import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

const Wrapper = styled.div.attrs((props) => {})`
  align-items: center;
  background-color: #fff4f4;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const Container = styled.div.attrs((props) => {})`
  align-items: center;
  background-color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: calc(120rem/1.6);
`;

const Heading = styled.h1.attrs((props) => {})`
  font-family: 'Dancing Script', cursive;
  font-size:  calc(3rem/1.6);
  margin:  calc(3rem/1.6)  calc(1rem/1.6)  calc(1rem/1.6)  calc(1rem/1.6);
`;

const Unordered = styled.ul.attrs((props) => {})``;
const ListItem = styled.li.attrs((props) => {})`
  font-size: calc(1.5rem/1.6);
  margin: calc(1rem/1.6);

  :last-child {
    margin-bottom: calc(3rem/1.6);
  }
`;

const Help = (props) => {
  return (
    <Wrapper>
      <Container>
        <Heading>Help Components</Heading>
        <Unordered>
          <ListItem>
            <Link to="/">Help</Link>
          </ListItem>
          {props.pages.map((page, num) => {
            return (
              <ListItem key={`${page.id}`}>
                <Link to={`/article/${page.id}`}>{`article/${page.id}`}</Link>
              </ListItem>
            );
          })}
        </Unordered>
      </Container>
    </Wrapper>
  );
};

const mapStateToProps = (state) => {
  return {
    pages: state.pages,
  };
};

export default connect(mapStateToProps)(Help);
export { Help };
