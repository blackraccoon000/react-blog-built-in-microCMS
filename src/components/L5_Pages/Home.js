import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import ArticleCard from '../L3_Organisms/ArticleCard';

const Wrapper = styled.div.attrs((props) => {})`
  align-items: center;
  background-color: #fff4f4;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

const Container = styled.div.attrs((props) => {})`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const Heading = styled.h1.attrs((props) => {})`
  font-family: 'Dancing Script', cursive;
  font-size: calc(30rem / 16);
  margin: calc(30rem / 16) calc(10rem / 16) calc(10rem / 16) calc(10rem / 16);
`;

const Unordered = styled.ul.attrs((props) => {})`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: calc(20rem / 16) auto;
`;

const ListItem = styled.li.attrs((props) => {})`
  font-size: calc(15rem / 16);
  margin: calc(10rem / 16);

  :last-child {
    margin-bottom: calc(30rem / 16);
  }
`;

const Home = (props) => {
  return (
    <Wrapper>
      <Container>
        <Unordered>
          {props.pages.map((page, num) => {
            return (
              <ListItem key={`${page.id}`}>
                <ArticleCard
                  src={page.thumbnail.url}
                  title={page.title}
                  link={`/article/${page.id}`}
                  createdAt={page.createdAt}
                  updatedAt={page.updatedAt}
                />
              </ListItem>
            );
          })}
        </Unordered>
      </Container>
    </Wrapper>
  );
};

const mapStateToProps = (state) => {
  return {
    pages: state.pages,
  };
};

export default connect(mapStateToProps)(Home);
export { Home };
