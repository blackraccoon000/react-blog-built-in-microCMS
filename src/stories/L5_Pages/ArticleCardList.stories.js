import React from 'react';
import { ArticleCardList } from '../../components/L5_Pages/ArticleCardList';

export default {
  title: 'L5_Pages/ArticleCardList',
  component: ArticleCardList,
};

const Template = (args) => <ArticleCardList {...args} />;
export const Default = Template.bind({});
Default.args = {};
