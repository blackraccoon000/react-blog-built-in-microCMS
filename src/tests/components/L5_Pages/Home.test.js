import React from 'react';
import { render } from '@testing-library/react';
import { Home } from '../../../components/L5_Pages/Home';
import {
  createDataOne,
  createDataTwo,
  createDataThree,
  createDataFour,
  createDataFive,
} from '../../fixtures/createData';

let wrapper;

const blogData = [
  createDataOne,
  createDataTwo,
  createDataThree,
  createDataFour,
  createDataFive,
];

beforeEach(() => {
  wrapper = render(<Home blogData={blogData} />);
});

it('Home Components Render', () => {
  expect(wrapper.container).toMatchSnapshot();
});
