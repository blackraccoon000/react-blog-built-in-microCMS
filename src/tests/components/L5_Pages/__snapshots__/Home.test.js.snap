// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components Render 1`] = `
<div>
  <div
    class="Home__Wrapper-sc-1yza5ij-0 eETUCZ"
  >
    <div
      class="Home__Container-sc-1yza5ij-1 eRfCXO"
    >
      <h1
        class="Home__Heading-sc-1yza5ij-2 wJefj"
      >
        Home Components
      </h1>
      <ul
        class="Home__Unordered-sc-1yza5ij-3"
      >
        <li
          class="Home__ListItem-sc-1yza5ij-4 jgCtOG"
        >
          first blog
           / 
          1 create blog now one
        </li>
        <li
          class="Home__ListItem-sc-1yza5ij-4 jgCtOG"
        >
          second blog
           / 
          2 create blog now two
        </li>
        <li
          class="Home__ListItem-sc-1yza5ij-4 jgCtOG"
        >
          third blog
           / 
          3 create blog now three
        </li>
        <li
          class="Home__ListItem-sc-1yza5ij-4 jgCtOG"
        >
          Forth blog
           / 
          4 create blog now four
        </li>
        <li
          class="Home__ListItem-sc-1yza5ij-4 jgCtOG"
        >
          Fifth blog
           / 
          5 create blog now five
        </li>
      </ul>
    </div>
  </div>
</div>
`;
